devtools/compat/pyhelpers/git.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python3
+# Copyright (c) The Libra Core Contributors
+# SPDX-License-Identifier: Apache-2.0
+from .cli import execute_cmd_with_text_output
+
+# length of git hash prefix
+SHORT_LEN = 7
+TAG_PREFIX = "master_"
```

Comment:
Heads up - this image may not exist in ECR.

ATM we don't build the full set of images for every commit. 
- LBT builds validator and safety-rules at the tip of the auto branch when a PR tries to land, and they use different tags (https://github.com/libra/libra/blob/ba042341101cfec720f6ec46992bf0bb3c99bbb1/.github/workflows/land-blocking.yml#L38) to distinguish images in trunk.   
- Continuous push builds every 3 hrs from the tip of then master. Only that commit has the full set of images and with the master_xxx tag.

Reply:
Ah right, thanks for catching that. I'll only need safety-rules and validator images for now, so I'll borrow the LBT `land_*` ones.

Reply:
Also doesn't look like we hold onto those images for too long. Shouldn't be too big of an issue for now but might limit option to test for example: HEAD vs HEAD~15. I've added a basic check for if the image exists before launching the test.